### ISSUE TITLE
 Group not support MultiInstanceAssociation

### ISSUE THREAD

**Issue by nechry** [OpenZWave/python-openzwave#23]
hi,

the manager.pxd 
https://github.com/OpenZWave/python-openzwave/blob/8515406322ff978ae1d1e9457213e4b72e2b9e35/src-lib/libopenzwave/manager.pxd#L170
not fully implement the CC MultiInstanceAssociation. we can specify a instance of desired node target: https://github.com/OpenZWave/open-zwave/blob/1c64a40037bdb8207c0afa6bd10be8ac3d3327fd/cpp/src/Manager.h#L1564
https://github.com/OpenZWave/open-zwave/blob/f32e84b7ace5ae929153c1897f52a4c44b5cecd7/cpp/src/Group.h#L84

same for Remove methods.

**Comment by bibi21000** [comment]
Fixed

**Comment by nechry** [comment]
I don't see the commit for this issue.
I work also on this see:
https://github.com/nechry/python-openzwave/commit/5a927e91bb6790366b0012448c01785c30f0110b

I found also another issue in group discovery
https://github.com/nechry/python-openzwave/commit/3f9b82b54c7a56458368491d0d715119b002a1b6
the index are not always is spet of one, is like parameter 1, 2, 3, 5, 6

**Comment by nechry** [comment]
je peux faire un PR ou tu prends et refactoring a ton gout si jamais

**Comment by nechry** [comment]
Thank, you will also check for MultiInstanceAssociation ? 
my code work and is tested. But, the only thing is, I don't yet updated the group class to include the instance property.

**Comment by nechry** [comment]
see for group discovery
<img width="1325" alt="screen shot 2016-04-18 at 22 41 48" src="https://cloud.githubusercontent.com/assets/307966/14619242/d5436218-05b6-11e6-95f1-987905f9292d.png">

thank's

**Comment by bibi21000** [comment]
Don't know what is your screen shot ?
Not tested my code but it's only a loop update. Of course, Errare humanum
est ;)
But will surely test it before releasing unless someone else will do it.
Update the group python class is the simplest things to do. It should be an
interresting things to identify instances in associations, ... update the
pyozwan, ...
Looking at your code : why 100 ? groups are coded with an int8 ... so 256
possible values ?
Looking at code, it seems that the group 255 is used sometimes ?

2016-04-18 22:43 GMT+02:00 Jean-François Auger notifications@github.com:

**Comment by nechry** [comment]
Hi @bibi21000 the screen is just to show you, results with a real device. actually the only one I know with this index irregular step. so is the test and is work. and the code still work with other devices to.

why 100?, yes when I wrote the patch I just set a upper boundary. 
ozw now provide in xmltest a case to avoid a device with MaxAssociation > of real group number association, normally the upper bound will never reaches. 

I never see a group 255, but I don't know every line of code too.

Just to return to this issue, I have to reopen a new issue for the MultiInstanceAssociation support? propose a PR with my actual implementation ? or you prefer do it by yourself. Or I really mist something and you already implemented.

many thanks
